test/apiv2/80-systemd.at

```diff
@@ -0,0 +1,62 @@
+# -*- sh -*-
+#
+# API service under socket activation
+#
+
+## Verify containers created under API service are properly configured
+## See https://bugzilla.redhat.com/show_bug.cgi?id=2052697
+
+set -x
+
+ACTIVATE=/usr/bin/systemd-socket-activate
+if ! [[ -x $ACTIVATE ]]; then
+    echo "skip: $(basename $0) tests require $ACTIVATE" >&2
+    return
+fi
+
+# systemd-socket-activate does not support DNS lookups
+ALTHOST=127.0.0.1
+
+ACTIVATE_PORT=$(random_port)
+if [[ $PORT == $ACTIVATE_PORT ]]; then
+    die "Duplicate port($PORT) returned for socket activation tests"
+fi
+
+$ACTIVATE --listen=$ALTHOST:$ACTIVATE_PORT $PODMAN_BIN \
+    --root $WORKDIR/server_root \
+    --syslog=true \
+    system service \
+    --time 0 \
+    &> >(tee $WORKDIR/activate_server.log) &
+ACTIVATE_PID=$!
+
+ACTIVATE_URL=http://$ALTHOST:$ACTIVATE_PORT/v4.0.0/libpod
+
+t POST "$ACTIVATE_URL/images/pull?reference=$IMAGE&quiet=true" 200
+
+t POST "$ACTIVATE_URL/containers/create" \
+    Image=$IMAGE \
+    Entrypoint='["top"]' \
+    Terminal=True \
+    201 \
+    .Id~[0-9a-f]\\{64\\}
+cid=$(jq -r '.Id' <<<"$output")
+t POST "$ACTIVATE_URL/containers/$cid/start" 204
+
+t GET "$ACTIVATE_URL/containers/$cid/json" 200 \
+      .State.Running=true
+
+# Emulate 'systemd stop podman.service'
+kill -TERM $ACTIVATE_PID
+wait -n $ACTIVATE_PID || rc=$?
```

Comment:
`-n` isn't needed here, hence it's misleading to a future maintainer.

I wonder if this is the cause of the hang?

Reply:
I added timeout to see.